From fbb35003a13225abcfd5ccac42873f3cada46b2a Mon Sep 17 00:00:00 2001
From: Matt Mower <mowerm@gmail.com>
Date: Tue, 4 Nov 2014 21:08:10 -0600
Subject: [PATCH 10/44] recovery: Allow device-specific recovery modules

Provide a means to include device-specific recovery modules that need to
be built (i.e. not pre-built). A list of recovery modules can be
specified by a device with:

TARGET_RECOVERY_DEVICE_MODULES := rec_mod_one rec_mod_two

Change-Id: Ibd19a71318863461d472471a4f7cfddfb1b9ae0d
---
 Android.mk | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Android.mk b/Android.mk
index 90d3b47c..5cc1a6c6 100644
--- a/Android.mk
+++ b/Android.mk
@@ -283,6 +283,10 @@ RECOVERY_TOOLS := \
     $(FILESYSTEM_TOOLS)
 LOCAL_POST_INSTALL_CMD := $(hide) $(foreach t,$(RECOVERY_TOOLS),ln -sf ${LOCAL_MODULE} $(LOCAL_MODULE_PATH)/$(t);)
 
+ifneq ($(TARGET_RECOVERY_DEVICE_MODULES),)
+    LOCAL_REQUIRED_MODULES += $(TARGET_RECOVERY_DEVICE_MODULES)
+endif
+
 include $(BUILD_EXECUTABLE)
 
 # mkshrc
-- 
2.17.1

